### DISCUSSION TITLE
Is there a way to see which import is unused?

### DISCUSSION THREAD

**Discussion by Austinito** [scalameta/metals#4915] [Q&A]
If I have an import statement with multiple items imported and only one is not being used, then my diagnostics will show which line(s) the issue is occuring, but I'm not able to determine which import is causing the issue.

```scala
import scala.annotation.{nowarn, showAsInfix}

@nowarn
class Example(unusedVar: String)
```

sbt compile will show the carrot under showAsInfix, but my Diagnostics will highlight the entire line and I'm unable to determine based on that.

Is there a way do determine which one is causing the issue?

**Comment by tgodzik** [comment]
What build server are you using? It seems to work correctly when testing in scala-cli, so laso Bloop should work. Also what Scala version are you using?

**Reply by Austinito** [reply]
Thanks for the reply:
I ran Metals Doctor and got the following output:
```md
## Metals Info
  - Metals Server version: 0.11.8
  - Build server currently being used is sbt v1.5.7.
  - Metals Java: 11.0.15 from BellSoft

## Build Targets

### root
  - target type: Scala 2.13.8
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠️ 
```

**Reply by tgodzik** [reply]
Looks like an issue with sbt. It seems to be sending the full line despite being able to show exact point in the console. @adpi2 any idea what might be going on?

**Reply by Austinito** [reply]
I also switch over to the bloop build server, and I'm seeing the same issue but I am compiling with the "-Werror" argurment -- not sure if that's what's causing the issue.

edit: Actually it's not "-Werror", it's "-Xfatal-warnings"

**Reply by tgodzik** [reply]
This might be a smalll issue somewhere I think (or at least I hope)

**Reply by adpi2** [reply]
Thanks for pinging me. I am working on it and let you know if I find an easy fix.

**Comment by adpi2** [comment]
From the example:

```scala
package example

import scala.annotation.{nowarn, showAsInfix}

@nowarn
class Example(unusedVar: String)
```

This is the position that the BSP reporter receives from the compiler:
```
startLine: 3
endLine: 3
startColumn: 0
endColumn: 45
pointer: 33
```

The problem here is that `startColumn` and `endColumn` span the entire line, rather than the single unused import `showAsInfix`. We could use the pointer as the start column but we would still miss the accurate end column.

It would make more sense to fix this in the compiler itself: https://github.com/scala/bug/issues/12734
